presto-native-execution/presto_cpp/main/http/filters/JwtTokenFilter.cpp

```diff
@@ -0,0 +1,154 @@
+    auto verifier = jwt::verify<jwt::traits::nlohmann_json>().allow_algorithm(
+        jwt::algorithm::hs256{std::string(
+            reinterpret_cast<char*>(secretHash.data()), secretHash.size())});
+
+    verifier.verify(decodedJwt);
+
+    auto subject = decodedJwt.get_subject();
+    // The nodeId of the requester is the subject. Check if it was set.
+    if (subject.empty()) {
+      std::error_code ec{jwt::error::token_verification_error::missing_claim};
+      throw jwt::error::token_verification_exception(ec);
+    }
+    // Passed the verification, move the message along.
+    Filter::onRequest(std::move(msg));
+  } catch (const jwt::error::token_verification_exception& e) {
+    sendUnauthorizedResponse();
+  } catch (const jwt::error::signature_verification_exception& e) {
+    sendUnauthorizedResponse();
+  } catch (const std::system_error& e) {
+    sendGenericErrorResponse();
+  }
+}
+
+void JWTTokenFilter::onBody(std::unique_ptr<folly::IOBuf> body) noexcept {
+  // If the request was rejected do no more processing
+  // and return.
+  // Otherwise, forward the request.
+  if (requestRejected_) {
+    return;
+  }
+  if (upstream_) {
+    // forward
+    upstream_->onBody(std::move(body));
+  }
+}
+
+void JWTTokenFilter::onEOM() noexcept {
+  // If the request was rejected do no more processing
+  // and return.
+  // Otherwise, forward the request.
+  if (requestRejected_) {
+    return;
+  }
+  if (upstream_) {
+    // forward
+    upstream_->onEOM();
+  }
+}
+
+void JWTTokenFilter::requestComplete() noexcept {
+  // If the request was rejected only cleanup
+  // needs to be done.
+  // Otherwise, forward the request.
+  if (requestRejected_) {
+    delete this;
+    return;
+  }
+  if (upstream_) {
+    upstream_->requestComplete();
+  }
+}
+
+void JWTTokenFilter::onUpgrade(proxygen::UpgradeProtocol protocol) noexcept {
+  // If the request was rejected do no more processing
+  // and return.
+  // Otherwise, forward the request.
+  if (requestRejected_) {
+    return;
+  }
+  if (upstream_) {
+    upstream_->onUpgrade(protocol);
+  }
+}
+
+void JWTTokenFilter::onError(proxygen::ProxygenError err) noexcept {
+  // If onError is invoked before we forward the error.
+  if (upstream_) {
+    upstream_->onError(err);
+    upstream_ = nullptr;
+  }
+
+  delete this;
+}
+
+void JWTTokenFilter::sendGenericErrorResponse(void) {
+  upstream_->onError(proxygen::kErrorUnsupportedExpectation);
+
+  proxygen::ResponseBuilder(downstream_)
+      .status(kHttpInternalServerError, "Internal Server Error")
+      .sendWithEOM();
+  requestRejected_ = true;
+}
+
+void JWTTokenFilter::sendUnauthorizedResponse() {
+  upstream_->onError(proxygen::kErrorUnauthorized);
+
+  proxygen::ResponseBuilder(downstream_)
+      .status(kHttpUnauthorized, "Unauthorized")
+      .header(proxygen::HTTP_HEADER_WWW_AUTHENTICATE, kPrestoInternalBearer)
+      .sendWithEOM();
+  requestRejected_ = true;
+}
+
+} // namespace facebook::presto::http::filters
```

Comment:
Remove the last line.

Reply:
This means there is a new line missing at the end. All other files have one. I'm adding it.